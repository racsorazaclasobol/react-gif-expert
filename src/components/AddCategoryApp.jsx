import PropTypes from 'prop-types';
import { useState } from "react"

export const AddCategoryApp = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target }) => {
    setInputValue( target.value )
  }
  
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    setInputValue('');
    onNewCategory(inputValue.trim())
  }

  return (
    <>
      <form onSubmit={(event) => onSubmit(event)} aria-label="form">
        <input 
              type="text"
              placeholder="Buscar Gif"
              value={inputValue}   
              onChange = {(event) => onInputChange(event)}       
          />
      </form>
    </>
  )
}

AddCategoryApp.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
