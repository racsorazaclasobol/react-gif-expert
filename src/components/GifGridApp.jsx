import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItemApp } from "./GifGridItemApp";
import PropTypes from 'prop-types'

export const GifGridApp = ({category}) => {

    //custom hooks
    const { images, isLoading } = useFetchGifs(category)

  return (
  <>
      
      <h4>{category}</h4>

      {
        isLoading && ( <h2>Cargando...</h2> )
      }

      <div className="card-grid">
        { images.map( (image) => (
            //El {...image} sirve para propagar todas las properties dentro del elemento image
            <GifGridItemApp key={image.id} {...image} />
        )) 
        }
      </div>
  </>
)
}

GifGridApp.propTypes = {
  category: PropTypes.string.isRequired,
}