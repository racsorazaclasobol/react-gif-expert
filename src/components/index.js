

export * from './AddCategoryApp'
export * from './GifGridApp'
export * from './GifGridItemApp'
