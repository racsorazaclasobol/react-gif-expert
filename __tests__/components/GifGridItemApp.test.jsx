import { render, screen } from "@testing-library/react"
import { GifGridItemApp } from "../../src/components/GifGridItemApp"

describe('Pruebas en <GifGridItemApp />', () => {

    const title = "Luffy";
    const url = "https://luffy.com/luffy.jpg"

    test('should be do match with snapshot', () => { 
        
        const { container } = render( <GifGridItemApp title = {title} url = {url}/> )

        expect( container ).toMatchSnapshot()

     })

     test('should show the image with the url and title indicated ', () => { 

        render( <GifGridItemApp title = {title} url = {url}/> )
        //screen.debug();
        //console.log(screen.getByRole('img').src)
        //expect( screen.getByRole('img').src ).toBe( url );

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( title );

      })

      test('should show the title on the component', () => { 
        
        render( <GifGridItemApp title = {title} url = {url}/> )

        expect( screen.getByText( title ) ).toBeTruthy();

       })


} )