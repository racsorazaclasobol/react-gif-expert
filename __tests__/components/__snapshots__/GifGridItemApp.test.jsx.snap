// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItemApp /> should be do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy"
      src="https://luffy.com/luffy.jpg"
    />
    <p>
      Luffy
    </p>
  </div>
</div>
`;
